spring-shell2-core/src/main/java/org/springframework/shell2/Shell.java

```
@@ -113,8 +116,17 @@ public void run() throws IOException {
 				Method method = methodTarget.getMethod();
 
 				try {
-					Object[] args = resolveArgs(method, wordsForArgs);
-					validateArgs(args, methodTarget);
+					Map<MethodParameter, Object> resolveResult = resolveArgs(method, wordsForArgs);
+					Object[] args = IntStream.range(0, method.getParameterCount()).mapToObj(index -> {
+						MethodParameter methodParameter = Utils.createMethodParameter(method, index);
+						Object parameterResult = resolveResult.get(methodParameter);
+						if (!resolveResult.containsKey(methodParameter)
+								|| !(resolveResult.get(methodParameter) instanceof ValueResult)) {
```

Ah :) ok, this takes me to a different point.

To provide completion context I updated the Shell to try to resolve all parameters instead of stopping at the first failure. Reason for this is that Parameter n+1 could've been successfully resolved even if Parameter n fails.

However, for the resolve operation we want to throw the first exception found, right? (at least to keep the logic we had before).

I could though:
1. update the completion logic to ignore Exceptions and try to resolve all parameters
1. change the resolve logic back to what it had before, as in, fail at first Exception